packages/medusa/src/api/routes/store/carts/create-cart.ts

```diff
@@ -135,7 +136,7 @@ export default async (req, res) => {
             })
           await cartService
             .withTransaction(manager)
-            .addLineItem(cart.id, lineItem)
+            .addLineItem(cart.id, lineItem, validated.validate_sales_channels)
```

Comment:
question: If one of my previous comment is true, then we should just pass true or `{ validate_sales_channels: true }` as the store route must check the sales channels

Reply:
maybe use the flag guard:

```js
  await cartService
    .withTransaction(manager)
    .addLineItem(cart.id, lineItem, { validate_sales_channels: flagRouter.isFeatureEnabled(...) })
```